### PR TITLE
Pickup&Item API for Exiled 6

### PR DESCRIPTION
**Pull request by louis1706**

### CODE REVIEW

**Comment by joker-119** [comment]
> do not merge label?

Drafts are automatically assumed to be "Do not merge", the label will be added if this is converted from a draft into a ready PR before we start accepting breaking changes into the dev branch.

File: Exiled.API/Features/Items/Firearm.cs [hunk 1]
```diff
@@ -513,6 +522,41 @@ public void ClearPreferences(IEnumerable<Player> players)
                 ClearPreferences(player);
         }
 
+        /// <summary>
+        /// Spawns the item on the map.
+        /// </summary>
+        /// <param name="position">The location to spawn the item.</param>
+        /// <param name="rotation">The rotation of the item.</param>
+        /// <param name="identifiers">The attachments to be added.</param>
+        /// <param name="spawn">Whether the <see cref="Pickup"/> should be initially spawned.</param>
```
Comment (moddedmcplayer) [inline Exiled.API/Features/Items/Firearm.cs:531]: Why add a Spawn param on Spawn method?

Comment (louis1706) [reply Exiled.API/Features/Items/Firearm.cs:531]: I look this later i don't remember doing this probably was in a PR i have merged

Comment (IRacle1) [reply Exiled.API/Features/Items/Firearm.cs:531]: Yes, these methods have a lil different logic, it is better to name them `CreatePickup`

**Comment by louis1706** [comment]
It's really need seryously test for all the API and All Event

File: Exiled.API/Features/Pickups/Pickup.cs [hunk 2]
```diff
@@ -0,0 +1,349 @@
...
+
+        /// <summary>
+        /// Gets the <see cref="ItemType"/> of the item.
+        /// </summary>
+        public ItemType Type => Base.NetworkInfo.ItemId;
+
+        /// <summary>
+        /// Gets or sets a value indicating whether the pickup is locked (can't be picked up).
+        /// </summary>
+        public bool Locked
```
Comment (louis1706) [inline Exiled.API/Features/Pickups/Pickup.cs:151]: Islocked ?

File: Exiled.API/Features/Pickups/Pickup.cs [hunk 3]
```diff
@@ -0,0 +1,349 @@
...
+            {
+                Base.Rb.rotation = value;
+                Base.RefreshPositionAndRotation();
+            }
+        }
+
+        /// <summary>
+        /// Gets a value indicating whether this pickup is spawned.
+        /// </summary>
+        public bool Spawned { get; internal set; }
```
Comment (louis1706) [inline Exiled.API/Features/Pickups/Pickup.cs:221]: IsSpawned

**Review by louis1706 — COMMENTED** [review]
TODO List

**Pull request closed without merging**